x-pack/plugins/ml/public/timeseriesexplorer/timeseriesexplorer_controller.js

```diff
@@ -282,13 +285,26 @@ module.controller('MlTimeSeriesExplorerController', function (
     const detectorIndex = +$scope.detectorId;
     $scope.modelPlotEnabled = isModelPlotEnabled($scope.selectedJob, detectorIndex, $scope.entities);
 
+
     // Only filter on the entity if the field has a value.
     const nonBlankEntities = _.filter($scope.entities, (entity) => { return entity.fieldValue.length > 0; });
     $scope.criteriaFields = [{
       'fieldName': 'detector_index',
       'fieldValue': detectorIndex }
     ].concat(nonBlankEntities);
 
+    if ($scope.modelPlotEnabled === false &&
+      isSourceDataChartableForDetector($scope.selectedJob, detectorIndex) === false &&
+      nonBlankEntities.length > 0) {
+      // For detectors where model plot has been enabled with a terms filter and the
+      // selected entity(s) are not in the terms list, indicate that data cannot be viewed.
+      $scope.hasResults = false;
+      $scope.loading = false;
+      $scope.dataNotChartable = true;
+      $scope.$applyAsync();
```

Comment:
it doesn't look like a promise is being used in this code, is this `$applyAsync` needed?

Reply:
Yes this `$applyAsync` is needed. When you are viewing an entity which triggers this block,  if you then hard refresh the page, the `$applyAsync` is needed to get the view to update correctly and show the correct error message on initial load.